app/models/compute_resources/foreman/model/openstack.rb

```diff
@@ -336,5 +340,27 @@ def allocate_address(network)
       logger.warn "failed to allocate ip address for network #{network}: #{e}"
       raise e
     end
+
+    def process_fixed_ips(args)
+      interfaces_attrs = args.delete(:interfaces_attributes)
+      return unless interfaces_attrs&.any?
+
+      # Sort interface attributes by index to match nics order
+      sorted_interfaces = interfaces_attrs.sort_by { |index, _| index.to_i }.map(&:last)
+
+      # Update nics with fixed IPs based on position
+      args[:nics].each_with_index do |nic, index|
+        next unless nic.is_a?(Hash) && nic['net_id']
```

Comment:
This could be filtered out at the begining of the method so we can skip the ordering and everything:

```ruby
nics = args[:nics].reject_if ....
return if nics.empty?
```